### ISSUE TITLE
Sad error reporting: panic

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/bazel-watcher#43]
Built from head
https://github.com/bazelbuild/bazel-watcher/commit/db7d755977e8619642ef1f223dcfa61628dd49c4

```
$ ibazel test examples/ts
State: QUERY
Querying for BUILD files...
Error running Bazel exit status 7
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x40eb463]

goroutine 1 [running]:
main.(*IBazel).queryForSourceFiles(0xc42009e000, 0xc42001e120, 0x22, 0x0, 0xc4200c4000, 0xc4200c6000)
	ibazel/ibazel.go:316 +0x1b3
main.(*IBazel).watchFiles(0xc42009e000, 0xc42001e120, 0x22, 0xc420066060)
	ibazel/ibazel.go:345 +0x5a
main.(*IBazel).iteration(0xc42009e000, 0x413a810, 0x4, 0xc420057e18, 0xc4200483b0, 0x1, 0x1, 0x7fff5fbffa04, 0xb)
	ibazel/ibazel.go:204 +0x91e
main.(*IBazel).loop(0xc42009e000, 0x413a810, 0x4, 0xc420057e18, 0xc4200483b0, 0x1, 0x1, 0xc420057e20, 0xc4200100e0)
	ibazel/ibazel.go:176 +0xf6
main.(*IBazel).Test(0xc42009e000, 0xc4200483b0, 0x1, 0x1, 0x1, 0x1)
	ibazel/ibazel.go:168 +0x80
main.handle(0xc42009e000, 0x7fff5fbff9ff, 0x4, 0xc4200100e0, 0x1, 0x1)
	ibazel/main.go:110 +0x22e
main.main()
	ibazel/main.go:99 +0x1bf
alexeagle-macbookpro:rules_karma alexeagle$ bazel build examples/ts
ERROR: error loading package 'examples/ts': Encountered error while reading extension file 'defs.bzl': no such package '@build_bazel_rules_typescript//': /private/var/tmp/_bazel_alexeagle/94bb2099ab3cc28ce303572fe6ffae30/external/build_bazel_rules_typescript must be an existing directory.
INFO: Elapsed time: 0.125s
```

**Comment by achew22** [comment]
Thanks for the bug report. Looks like I assumed that bazel would always return to me a proto when I ask for --output=proto. It looks like it can also send back a string of text. I should probably catch the error and trigger a `bazel build` with WriteToStdOut to get the best error reporting.

**Comment by dougkoch** [comment]
This bug is reproducible with commits up to cd96809, but @achew22 fixed it in #48 (e8cbba3). The only reason the bazel query didn't return a proto in @alexeagle's example is that the query failed. It then segfaulted when queryForSourceFiles() ignored the query failure and attempted to pull the targets out of the proto anyway. After #48, queryForSourceFiles() now returns immediately if the query fails, so this segfault is no longer possible and I'm closing this issue.
